kedro/framework/cli/micropkg.py

```diff
@@ -0,0 +1,916 @@
+from textwrap import indent
+from typing import Any, Iterable, List, NamedTuple, Optional, Set, Tuple, Union
+from zipfile import ZipFile
+
+import click
+import pkg_resources
+from rope.base.project import Project
+from rope.contrib import generate
+from rope.refactor.move import MoveModule
+from rope.refactor.rename import Rename
+from setuptools.dist import Distribution
+
+from kedro.framework.cli.utils import (
+    KedroCliError,
+    _clean_pycache,
+    _get_requirements_in,
+    call,
+    command_with_verbosity,
+    env_option,
+    python_call,
+)
+from kedro.framework.project import settings
+from kedro.framework.startup import ProjectMetadata
+
+_SETUP_PY_TEMPLATE = """# -*- coding: utf-8 -*-
+from setuptools import setup, find_packages
+
+setup(
+    name="{name}",
+    version="{version}",
+    description="Modular pipeline `{name}`",
+    packages=find_packages(),
+    include_package_data=True,
+    package_data={package_data},
+    install_requires={install_requires},
+)
+"""
+
+
+class PipelineArtifacts(NamedTuple):
+    """An ordered collection of source_path, tests_path, config_paths"""
+
+    pipeline_dir: Path
+    pipeline_tests: Path
+    pipeline_conf: Path
+
+
+def _assert_pkg_name_ok(pkg_name: str):
+    """Check that python package name is in line with PEP8 requirements.
+
+    Args:
+        pkg_name: Candidate Python package name.
+
+    Raises:
+        KedroCliError: If package name violates the requirements.
+    """
+
+    base_message = f"`{pkg_name}` is not a valid Python package name."
+    if not re.match(r"^[a-zA-Z_]", pkg_name):
+        message = base_message + " It must start with a letter or underscore."
+        raise KedroCliError(message)
+    if len(pkg_name) < 2:
+        message = base_message + " It must be at least 2 characters long."
+        raise KedroCliError(message)
+    if not re.match(r"^\w+$", pkg_name[1:]):
+        message = (
+            base_message + " It must contain only letters, digits, and/or underscores."
+        )
+        raise KedroCliError(message)
+
+
+def _check_pipeline_name(ctx, param, value):  # pylint: disable=unused-argument
+    if value:
+        _assert_pkg_name_ok(value)
+    return value
+
+
+def _check_module_path(ctx, param, value):  # pylint: disable=unused-argument
+    if value and not re.match(r"^[\w.]+$", value):
+        message = "The pipeline location you provided is not a valid Python module path"
+        raise KedroCliError(message)
+    return value
+
+
+# pylint: disable=missing-function-docstring
+@click.group(name="Kedro")
+def micropkg_cli():  # pragma: no cover
+    pass
+
+
+@micropkg_cli.group()
+def micropkg():
+    """Commands for working with micro-packages."""
+
+
+@command_with_verbosity(micropkg, "pull")
+@click.argument("package_path", nargs=1, required=False)
+@click.option(
+    "--all",
+    "-a",
+    "all_flag",
+    is_flag=True,
+    help="Pull and unpack all pipelines in the `pyproject.toml` package manifest section.",
```

Comment:
This command is now not only for `pipelines` anymore right? Since this help message is user facing, I think we should update it to reflect the functionality. And the same with the help messages for the other CLI commands in this file.

Reply:
Agreed I will update all help messages to reflect this.